src/main/java/seedu/duke/Penalty.java

```diff
@@ -27,15 +27,9 @@ public static void executePenalty() {
      * @return The user's penalty direction as an integer (0, 1, or 2).
      */
     private static int getUserPenaltyDirection() {
-        int userDirection;
-        do {
-            System.out.print(PENALTY_PROMPT);
-            String directionString = Ui.IN.nextLine().trim();
-            userDirection = convertToShootDirection(directionString);
-            if (userDirection == -1) {
-                System.out.println("Invalid direction! Please enter 0, 1, or 2.");
-            }
-        } while (userDirection == -1);
+        System.out.print(PENALTY_PROMPT);
+        String directionString = Ui.IN.nextLine();
```

Comment:
This still needs changes. User's command is read in beginListening() method in Ui class.